modules/imgcodecs/src/grfmt_jpeg2000.cpp

```diff
@@ -476,6 +477,17 @@ bool  Jpeg2KEncoder::write( const Mat& _img, const std::vector<int>& )
     if( channels > 3 || channels < 1 )
         return false;
 
+    double target_compression_rate = 1.0;
+    for( size_t i = 0; i < params.size(); i += 2 )
```

Comment:
there should be check that `params.size() % 2 == 0`

Reply:
Thank you for your comments!
Indeed a valid point!
I will add a check, if that's still required after resolving https://github.com/opencv/opencv/pull/13847#discussion_r258039971.